app/models/piece.rb

```diff
@@ -8,77 +8,87 @@ class Piece < ActiveRecord::Base
-    for i in (x+1...current_row) do
-      if self.game.piece_exists?(i, y)
-        return true
+    # check if piece is blocking the path in negative vertical direction
+    if current_col == col && row < current_row
+      for i in (row+1...current_row) do
+        if self.game.piece_exists?(i, col)
+          return true
+        end
       end
     end
-  end
 
-  # check if piece is blocking the path in positive horizontal direction
-  if current_row == x && y > current_col
-    for i in (current_col+1...y) do
-      if self.game.piece_exists?(x, i)
-        return true
+    # check if piece is blocking the path in positive horizontal direction
+    if current_row == row && col > current_col
+      for i in (current_col+1...col) do
+        if self.game.piece_exists?(row, i)
+          return true
+        end
       end
     end
-  end
 
-  # check if piece is blocking the path in negative horizontal direction
-  if current_row == x && y < current_col
-    for i in (y+1...current_col) do
-      if self.game.piece_exists?(x, i)
-        return true
+    # check if piece is blocking the path in negative horizontal direction
+    if current_row == row && col < current_col
+      for i in (col+1...current_col) do
+        if self.game.piece_exists?(row, i)
+          return true
+        end
       end
     end
-  end
 
-  return false
-end
+    return false
+  end
 
-def puts_piece
-  puts "I am a piece"
-end
+  def puts_piece
+    puts "I am a piece"
+  end
 
-def occupied?(x, y)
-  game.pieces.where(row_position: x, col_position: y).present?
-end
+  def in_boundaries?(row, col)
+    col >= 0 && col <=7 && row >= 0 && row <= 7
+  end
 
-def in_boundaries?(row, col)
-  return col >= 0 && col <=7 && row >= 0 && row <= 7
-end
+  def legal_horiz_move?(row, col)
+    #row does NOT change
+    row >= 0 && row <= 7 && col >= 0 && col <= 7 && (self.col_position - col_position) == 0
+    #could we use self.in_boundaries && (self.col_position -col_position) == 0?
+  end
 
-def legal_horiz_move?(row, col)
-  return row >= 0 && row <= 7 && col >= 0 && col <= 7 && (self.col_position - col_position) == 0
-end
+  def legal_vert_move?(row, col)
+    #col does NOT change
+    row >= 0 && row <= 7 && col >= 0 && col <= 7 && (self.row_position - row_position) == 0
+    #could we use self.in_boundaries && (self.row_position - row_position) == 0
+  end
 
-def legal_vert_move?(row, col)
-  return row >= 0 && row <= 7 && col >= 0 && col <= 7 && (self.row_position - row_position) == 0
-end
+  def legal_diag_move?(row, col)
+    #abs val row must equal abs value col
+    (self.row_position - row_position).abs == (self.col_position - col_position).abs
+  end
 
-def legal_diag_move?(row, col)
-  #row >= 0 && row <= 7 && col >= 0 && col <= 7
-  (self.row_position - row_position).abs == (self.col_position - col_position).abs
-end
+  def piece_exists?(row, col)
+    Piece.where(game_id: self.id, row_position: row, col_position: col).exists?
+  end
```

Comment:
use pieces.where( ) and don't add game_id in the params.  Rails will take care of the association.